src/CodeReport/Disassembler.cpp

```diff
@@ -24,20 +40,41 @@ void Disassembler::Disassemble(const void* machine_code, size_t size, uint64_t a
   AddLine(absl::StrFormat("Platform: %s",
                           is_64bit ? "X86 64 (Intel syntax)" : "X86 32 (Intel syntax)"));
   err = cs_open(arch, mode, &handle);
-  if (err) {
+  if (err != CS_ERR_OK) {
     AddLine(absl::StrFormat("Failed on cs_open() with error returned: %u", err));
     return;
   }
 
+  cs_option(handle, CS_OPT_DETAIL, CS_OPT_ON);
+
   count = cs_disasm(handle, static_cast<const uint8_t*>(machine_code), size, address, 0, &insn);
 
   if (count) {
     size_t j;
 
     for (j = 0; j < count; j++) {
-      AddLine(
-          absl::StrFormat("0x%llx:\t%-12s %s", insn[j].address, insn[j].mnemonic, insn[j].op_str),
-          insn[j].address);
+      if (IsCallInstruction(insn[j])) {
+        uint64_t callee_address = 0;
+        const orbit_client_protos::FunctionInfo* callee = nullptr;
+        if (absl::numbers_internal::safe_strtou64_base(insn[j].op_str, &callee_address, 16)) {
+          callee = orbit_client_data::FindFunctionByAddress(process, module_manager, callee_address,
+                                                            false);
+        }
+        if (callee != nullptr) {
+          AddLine(absl::StrFormat("0x%llx:\t%-12s %s (%s)", insn[j].address, insn[j].mnemonic,
+                                  insn[j].op_str,
+                                  orbit_client_data::function_utils::GetDisplayName(*callee)),
+                  insn[j].address);
+        } else {
+          AddLine(absl::StrFormat("0x%llx:\t%-12s %s ([%s])", insn[j].address, insn[j].mnemonic,
```

Comment:
What's the rationale for having two types of brackets? `([???])` Are we rendering it like this in the data views?

Reply:
I thought we render it like this in the other views, but that's not true. So changed it to `(???)`.

Reply:
The regular expression won't work anymore, but consider including only that border case "???" directly (maybe with a comment to explaining that we are using "???" in that case?)

Reply:
I also updated the regular expression, and tested that it still works. Do I miss something?

Reply:
Sorry, I was reading the old code.